### PR TITLE
Fix issue where we might run code in the lightbulb post-disposal

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs [hunk 1]
```diff
@@ -153,7 +153,8 @@ public bool TryGetTelemetryId(out Guid telemetryId)
             private void OnTextViewClosed(object sender, EventArgs e)
                 => Dispose();
 
-            public async Task<ISuggestedActionCategorySet?> GetSuggestedActionCategoriesAsync(ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range, CancellationToken cancellationToken)
+            public async Task<ISuggestedActionCategorySet?> GetSuggestedActionCategoriesAsync(
+                ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range, CancellationToken cancellationToken)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs:157]: view with whitespace off.

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs [hunk 2]
```diff
@@ -186,7 +187,7 @@ private void OnTextViewClosed(object sender, EventArgs e)
                 cancellationToken = linkedTokenSource.Token;
 
                 // Kick off the work to get errors.
-                var errorTask = GetFixLevelAsync();
+                var errorTask = GetFixLevelAsync(document, range, fallbackOptions, cancellationToken);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs:190]: made normal methods, so they don't capture 'state' accidetnally.

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs [hunk 3]
```diff
@@ -207,28 +208,40 @@ private void OnTextViewClosed(object sender, EventArgs e)
...
 
-                    return null;
+                foreach (var order in Orderings)
+                {
+                    var priority = TryGetPriority(order);
+                    Contract.ThrowIfNull(priority);
+                    var priorityProvider = new SuggestedActionPriorityProvider(priority.Value, lowPriorityAnalyzers);
+
+                    var result = await GetFixCategoryAsync(
+                        priorityProvider).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs:237]:
```suggestion
                    var result = await GetFixCategoryAsync(priorityProvider).ConfigureAwait(false);
```

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs [hunk 4]
```diff
@@ -207,28 +208,39 @@ private void OnTextViewClosed(object sender, EventArgs e)
...
+
+                    var result = await GetFixCategoryAsync(priorityProvider).ConfigureAwait(false);
+                    if (result != null)
+                        return result;
                 }
 
-                async Task<string?> GetFixCategoryAsync(ICodeActionRequestPriorityProvider priorityProvider)
+                return null;
+
+                async Task<string?> GetFixCategoryAsync(SuggestedActionPriorityProvider priorityProvider)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs:243]:
```suggestion
                async Task<string?> GetFixCategoryAsync(ICodeActionRequestPriorityProvider priorityProvider)
```

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**